smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/RequestTestEndpointResolverMiddleware.kt

```diff
@@ -26,6 +26,8 @@ class RequestTestEndpointResolverMiddleware(private val model: Model, private va
             ClientRuntimeTypes.Core.SdkError
         ) {
             writer.write("input.withPath(context.getPath())")
+            writer.write("let host = \"\\(context.getHostPrefix() ?? \"\")\\(context.getHost() ?? \"\")\"")
+            writer.write("input.withHost(host)")
```

Comment:
To keep things simple, i've added these two lines -- in the future, we can work on unifying the endpoint resolver to the request tests